ops/terraform/services/eft/main.tf

```diff
@@ -26,11 +34,64 @@ locals {
     for key, value in local.sensitive_service_map : split("/", key)[5] => value
   }
 
+  data_env      = lookup(local.nonsensitive_common_config, "ephemeral_environment_seed", local.env)
+  kms_key_alias = local.nonsensitive_common_config["kms_key_alias"]
+  vpc_name      = local.nonsensitive_common_config["vpc_name"]
+
   subnet_ip_reservations = jsondecode(
     local.sensitive_service_config["subnet_to_ip_reservations_nlb_json"]
   )
+  host_key              = local.sensitive_service_config["sftp_transfer_server_host_private_key"]
+  eft_user_sftp_pub_key = local.sensitive_service_config["sftp_eft_user_public_key"]
+  eft_user_username     = local.sensitive_service_config["sftp_eft_user_username"]
+
+  kms_key_id     = data.aws_kms_key.cmk.arn
+  sftp_port      = 22
+  logging_bucket = "bfd-${local.data_env}-logs-${local.account_id}"
+
+  # For some reason, the transfer server endpoint service does not support us-east-1b and instead
+  # opts to support us-east-1d. In order to enable support for this sub-az in the future
+  # automatically (if transfer server VPC endpoints begin to support 1c), we filter our desired
+  # subnets against the supported availability zones taking only those that belong to supported azs
```

Comment:
Huh. I'm curious what the behavior is here... does `aws_vpc_endpoint.this` generally fail to create when `use1-az4` is included? Does it yield a more specific error, e.g. _unsupported availability zone_ or something?

Reply:
Yeah, when trying to specify the subnet associated with that availability zone we get the following error:

```
The VPC endpoint service com.amazonaws.us-east-1.transfer.server does not support the availability zone of the subnet: <us-east-1b subnet ID>
```

For whatever reason `com.amazonaws.us-east-1.transfer.server` doesn't support `us-east-1b`. As far as I understand it, the VPC Endpoint needs to match its service name with the service its acting as an endpoint for, so I don't think we can get around this limitation.

Reply:
@mjburling I've been thinking about this, and I believe if we try to reach the SFTP Transfer Server through the NLB's `us-east-1b` Network Interface it will never connect (because there is no corresponding `us-east-1b` VPC Endpoint). Should we enable Cross-Zone Load Balancing to remedy this?

Reply:
> @mjburling I've been thinking about this, and I believe if we try to reach the SFTP Transfer Server through the NLB's `us-east-1b` Network Interface it will never connect (because there is no corresponding `us-east-1b` VPC Endpoint). Should we enable Cross-Zone Load Balancing to remedy this? Or, should we also filter the NLB's subnets to match the subnets the VPC Endpoint belongs to?

This is all rather intuitive, but the problem didn't even cross my mind when you mentioned it before. I'm not clear on why us-east-1b is unavailable for this service and I think it might be a good idea to open a ticket with AWS to understand why this limitation exists. However, if we are provisioning endpoints in us-east-1b and sharing the us-east-1b homed ip address with the responsible teams, then us-east-1b endpoints must be functional. For this solution, we really do want the three-AZ redundancy. Cross AZ loadbalancing is probably fine, provided we keep an eye on any truly untoward costs that result from it.

Reply:
Sounds good, I'll enable cross AZ load balancing. Considering the files being transferred should be _relatively_ small, I don't expect we'll run into unexpectedly high cost doing so.

Thanks for taking a look into this.